src/components/Map.jsx

```diff
@@ -1,34 +1,53 @@
+import "maplibre-gl/dist/maplibre-gl.css";
 import T from "prop-types";
 import MapGl, { Layer, Source } from "react-map-gl/maplibre";
-import "maplibre-gl/dist/maplibre-gl.css";
-import bbox from "@turf/bbox";
+import { useState, useEffect } from "react";
 import { AttributionControl } from "react-map-gl";
+import bbox from "@turf/bbox";
+import { mapLayersAtom } from "../atoms";
+import { useAtomValue } from "jotai";
 
-function Map({ artifact, mini }){
-  const isStatic = mini; // if mini is true, the map is static
-  let viewState = {
+/**
+ * Map component
+ * Children are the layers to render on the map
+ */
+function Map() {
+  const [viewState, setViewState] = useState({
```

Comment:
I don't think we need to track the view state of the map. The map implementation itself does that for us, and it exposes all the methods we need to adjust the view. 

We can register a `ref` on the map and then use methods like `setZoom` or `fitBounds` to change the view if we need to. See [example here](https://visgl.github.io/react-map-gl/docs/api-reference/map#methods)